tests/Unit/FormRequests/AdminUpdateVoucherRequestTest.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+namespace Tests\Unit\FormRequests;
+
+use App\Http\Requests\AdminUpdateVoucherRequest;
+use App\Sponsor;
+use App\User;
+use App\Voucher;
+use Auth;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Validation\Validator;
+use Tests\StoreTestCase;
+
+class AdminUpdateVoucherRequestTest extends StoreTestCase
+{
+    use DatabaseMigrations;
+
+    /** @var Validator */
+    private $validator;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->validator = app()->get('validator');
+
+        $rangeCodes = [
+            'TST0101',
+            'TST0102',
+            'TST0103',
+            'TST0104',
+            'TST0105',
+        ];
+
+        // Make a sponsor to match
+        $sponsor = factory(Sponsor::class)->create(
+            ['shortcode' => 'TST']
+        );
+
+        // Make an error Sponsor
+        $err_sponsor = factory(Sponsor::class)->create(
+            ['shortcode' => 'ERR']
+        );
+
+        // TODO: Looks ike voucher transitioning might be dependent on trader users
```

Comment:
*ike

... and is that a problem? Since Admins need to be able to transition vouchers too?

Reply:
Currently, it is not a problem, due to pure accident of mapping; it is tech debt and needs fixing, as a separate card i'll make.